discovery_child_development/notebooks/03_prepare_training_data.py

```diff
@@ -0,0 +1,206 @@
+# %% [markdown]
+# # Prepare labelled data
+#
+# The steps are:
+# * Load the taxonomy from Google sheets; concepts metadata; abstracts
+# * Filter the taxonomy to just records that were marked relevant ("Y")
+# * Filter the concepts metadata to just concepts that were marked "Y" in the taxonomy
+# * Attach the filtered concepts metadata (and sub-categories) to the abstracts. If any abstracts were tagged with NO relevant concepts/subcategories, these will now be lost.
+# * Split and put aside a test dataset based on OpenAlex IDs
+
+# %%
+import pandas as pd
+from df2gspread import gspread2df as g2d
+from oauth2client.service_account import ServiceAccountCredentials
+import os
+import logging
+from sklearn.model_selection import train_test_split
+
+from nesta_ds_utils.loading_saving import S3
+
+from discovery_child_development import PROJECT_DIR, logging
+from discovery_child_development.utils import bigquery, labelling_utils
+from discovery_child_development.utils.io import import_config
+
+bigquery.find_credentials("GOOGLE_SHEETS_CREDENTIALS")
+
+GOOGLE_SHEETS_CREDENTIALS = os.path.join(
+    PROJECT_DIR, os.environ["GOOGLE_SHEETS_CREDENTIALS"]
+)
+
+S3_BUCKET = os.environ["S3_BUCKET"]
+
+PARAMS = import_config("config.yaml")
+
+CONCEPT_IDS = "|".join(PARAMS["openalex_concepts"])
+
+OUT_PATH = "data/openAlex/processed/"
+
+SEED = 42
+
+
+# %%
+# Helpful functions!
+
+
+# For cleaning sub-category labels
+def clean_string(s):
+    s = s.replace(
+        "?", ""
+    )  # Remove question marks (do this *before* removing trailing spaces)
+    s = s.strip()  # Remove leading/trailing whitespace
+    s = " ".join(s.split())  # Remove duplicate spaces
+    s = s.lower()  # Convert to lowercase
+    return s
+
+
+# For accessing data from Google Sheets
+def access_google_sheet(google_credentials_json, sheet_id, sheet_name):
+    # Define the scope for the Google Sheets API (we only want Google sheets)
+    scope = ["https://spreadsheets.google.com/feeds"]
+
+    # Authenticate using the credentials JSON file
+    credentials = ServiceAccountCredentials.from_json_keyfile_name(
+        google_credentials_json, scope
+    )
+
+    # Load the data into a pandas DataFrame
+    data = g2d.download(
+        sheet_id, sheet_name, credentials=credentials, col_names=True, row_names=True
+    )
+
+    return data
```

Comment:
I'd suggest factoring this out into a module (eg, perhaps makes sense to rename bigquery.py to google.py and keep all google-related utils there?)